src/t_stream.c

```diff
@@ -1783,34 +1804,166 @@ void streamPropagateConsumerCreation(client *c, robj *key, robj *groupname, sds
 #define STREAM_RWR_HISTORY (1<<2)       /* Only serve consumer local PEL. */
-size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end, size_t count, int rev, streamCG *group, streamConsumer *consumer, int flags, streamPropInfo *spi, unsigned long *propCount) {
+#define STREAM_RWR_CLAIMED (1<<3)       /* Only serve claimed entries from PEL. */
+size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end, size_t count, int rev, long long min_idle_time, streamCG *group, streamConsumer *consumer, int flags, streamPropInfo *spi, unsigned long *propCount) {
     void *arraylen_ptr = NULL;
     size_t arraylen = 0;
     streamIterator si;
     int64_t numfields;
     streamID id;
+    uint64_t keyBuf[3];
+    pelTimeKey timeKey;
     int propagate_last_id = 0;
     int noack = flags & STREAM_RWR_NOACK;
 
     if (propCount) *propCount = 0;
 
+    if(group && min_idle_time != -1) {
+        arraylen_ptr = addReplyDeferredLen(c);
+        /* Scan the group's pending entries list (PEL) to find messages that have been
+         * idle for at least min_idle_time milliseconds. The pel_by_time radix tree
+         * stores entries ordered by their last delivery timestamp, allowing us to
+         * efficiently iterate from oldest to newest.
+         *
+         * We collect eligible entries into a temporary list rather than processing
+         * them inline because:
+         * 1. We cannot safely modify a radix tree while iterating over it
+         * 2. The claiming process requires removing and re-inserting entries in
+         *    both pel_by_time and the consumer PELs
+         *
+         * The iteration can terminate early in two cases:
+         * 1. We find an entry that hasn't been idle long enough - due to time-based
+         *    ordering, all subsequent entries will be even newer
+         * 2. We've collected enough entries to satisfy the requested count limit */
+        list *eligible_pels = listCreate();
+        raxIterator ri;
+        raxStart(&ri, group->pel_by_time);
+        raxSeek(&ri, "^", NULL, 0);
+        while(raxNext(&ri)) {
+            pelTimeKey pelKey;
+            decodePelTimeKey(ri.key, &pelKey);
+            uint64_t idle = commandTimeSnapshot() - pelKey.delivery_time;
+            if(idle < (uint64_t)min_idle_time)
+                break;
+
+            /* Store a copy of the key for later processing */
+            pelTimeKey *keyCopy = zmalloc(sizeof(pelTimeKey));
+            memcpy(keyCopy, &pelKey, sizeof(pelTimeKey));
+            listAddNodeTail(eligible_pels, keyCopy);
+            
+            if (count && listLength(eligible_pels) >= count) break;
+        }
+        raxStop(&ri);
+
+        /* Process each eligible pending entry, claiming it for the current consumer.
+         * For each entry we:
+         * 1. Verify it still exists in the PEL (it may have been acknowledged meanwhile)
```

Comment:
Shouldn't we do that check inside the above loop? Unless I'm missing something - 
Say `count == N` and we have `N + M`(where M < N) entries with `idle time >= min_idle time`, but the first M have been acknowledged. We will put the first N inside `eligible_pels` but the first M  will be skipped.
Whereas we could have put the last N inside the `eligible_pels` list